core/shared/src/main/scala/io/circe/Json.scala

```diff
@@ -124,6 +125,27 @@ sealed abstract class Json extends Product with Serializable {
   final def spaces4: String = Printer.spaces4.pretty(this)
 
   /**
+    * Perform a deep merge of this JSON value with another JSON value.
+    *
+    * Objects are merged by key, values from the argument JSON take
+    * precedence over values from this JSON. Nested objects are
+    * recursed.
+    *
+    * Null, Array, Boolean, String and Number are treated as values,
+    * and values from the argument JSON completely replace values
+    * from this JSON.
+    */
+  def deepmerge(that: Json): Json =
+    Apply[Option].map2(asObject, that.asObject) { (lhs, rhs) =>
```

Comment:
No problem, I got it - will revert to pattern matching and push the change within the next hour or two.